evap/staff/views.py

```diff
@@ -905,17 +905,37 @@ def semester_questionnaire_assign(request, semester_id):
 
     if form.is_valid():
         for evaluation in evaluations:
-            if form.cleaned_data[evaluation.course.type.name]:
-                evaluation.general_contribution.questionnaires.set(form.cleaned_data[evaluation.course.type.name])
-            if form.cleaned_data["all-contributors"]:
+            general_questionnaires = list(form.cleaned_data[f"general-{evaluation.course.type.id}"])
+            contributor_questionnaires = list(
+                form.cleaned_data["all-contributors"] | form.cleaned_data[f"contributor-{evaluation.course.type.id}"]
+            )
+
+            if form.cleaned_data[f"general-{evaluation.course.type.id}"]:
+                evaluation.general_contribution.questionnaires.set(general_questionnaires)
+
+            if form.cleaned_data[f"contributor-{evaluation.course.type.id}"] or form.cleaned_data["all-contributors"]:
```

Comment:
Instead of reading relatively random-looking keys in `cleaned_data` again, can we use `general_questionnaires` and `contributor_questionnaires` please?